crda/remotematcher.go

```diff
@@ -0,0 +1,404 @@
+package crda
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/quay/zlog"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/label"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+)
+
+var (
+	_ driver.Matcher             = (*Matcher)(nil)
+	_ driver.RemoteMatcher       = (*Matcher)(nil)
+	_ driver.MatcherConfigurable = (*Matcher)(nil)
+)
+
+const (
+	// Bounded concurrency limit.
+	defaultBatchSize          = 100
+	defaultEndPoint           = "/api/v2/vulnerability-analysis"
+	defaultRequestConcurrency = 10
+	defaultURL                = "https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v2/"
+	defaultSource             = "clair-upstream"
+	defaultKey                = "user_key=9e7da76708fe374d8c10fa752e72989f"
```

Comment:
Isn't it defaultSource redundant? You should able to find the source using default user key from backend.
Also there is a `user agent` http header which can be used for this purpose.

Reply:
@arajkumar the purpose here is to track users.Since this there is a possiblity that others can also get and use the other key.
A dedicated source is hence required.